### PR TITLE
Prepare for EnergyPlus 23.1

### PR DESCRIPTION
**Pull request by Myoldmopar**
Pull request overview
---------------------
Some cleanups and final preparations for 23.1.
 - Only keeping transition programs back to 9.0.0 in the installer, this should help reduce the installer bloat.
 - Updated energyplus and transition EP-Launch workflows, mostly just cleanup

**NOTE: ENHANCEMENTS MUST FOLLOW A SUBMISSION PROCESS INCLUDING A FEATURE PROPOSAL AND DESIGN DOCUMENT PRIOR TO SUBMITTING CODE**

### Pull Request Author
Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
 - [ ] Title of PR should be user-synopsis style (clearly understandable in a standalone changelog context)
 - [ ] Label the PR with at least one of: Defect, Refactoring, NewFeature, Performance, and/or DoNoPublish
 - [ ] Pull requests that impact EnergyPlus code must also include unit tests to cover enhancement or defect repair
 - [ ] Author should provide a "walkthrough" of relevant code changes using a GitHub code review comment process
 - [ ] If any diffs are expected, author must demonstrate they are justified using plots and descriptions
 - [ ] If changes fix a defect, the fix should be demonstrated in plots and descriptions
 - [ ] If any defect files are updated to a more recent version, upload new versions here or on DevSupport
 - [ ] If IDD requires transition, transition source, rules, ExpandObjects, and IDFs must be updated, and add IDDChange label
 - [ ] If structural output changes, add to output rules file and add OutputChange label
 - [ ] If adding/removing any LaTeX docs or figures, update that document's CMakeLists file dependencies

### Reviewer
This will not be exhaustively relevant to every PR.
 - [ ] Perform a Code Review on GitHub
 - [ ] If branch is behind develop, merge develop and build locally to check for side effects of the merge
 - [ ] If defect, verify by running develop branch and reproducing defect, then running PR and reproducing fix
 - [ ] If feature, test running new feature, try creative ways to break it
 - [ ] CI status: all green or justified
 - [ ] Check that performance is not impacted (CI Linux results include performance check)
 - [ ] Run Unit Test(s) locally
 - [ ] Check any new function arguments for performance impacts
 - [ ] Verify IDF naming conventions and styles, memos and notes and defaults
 - [ ] If new idf included, locally check the err file and other outputs

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -24,6 +24,27 @@
 .eplaunch
 testfiles/EPLaunchRun_*/
 
+# EnergyPlus output files in the testsilfes folder
```
Comment (Myoldmopar) [inline .gitignore:27]: Just ignore a few EnergyPlus output files in the case of running inside the repo.

File: cmake/Install.cmake [hunk 2]
```diff
@@ -361,7 +361,6 @@ install(
 
 # TODO Remove version from file name or generate
 # These files names are stored in variables because they also appear as start menu shortcuts later.
-set(RULES_XLS Rules22-2-0-to-22-3-0.md)
```
Comment (Myoldmopar) [inline cmake/Install.cmake:364]: This isn't used, and was wrong anyway.  If we want to use variables, we should just use the raw E+ version variables.

File: idd/CMakeLists.txt [hunk 3]
```diff
@@ -1,14 +1,14 @@
...
+#install(FILES "V8-0-0-Energy+.idd" DESTINATION "PreProcess/IDFVersionUpdater")
+#install(FILES "V8-1-0-Energy+.idd" DESTINATION "PreProcess/IDFVersionUpdater")
+#install(FILES "V8-2-0-Energy+.idd" DESTINATION "PreProcess/IDFVersionUpdater")
+#install(FILES "V8-3-0-Energy+.idd" DESTINATION "PreProcess/IDFVersionUpdater")
+#install(FILES "V8-4-0-Energy+.idd" DESTINATION "PreProcess/IDFVersionUpdater")
+#install(FILES "V8-5-0-Energy+.idd" DESTINATION "PreProcess/IDFVersionUpdater")
+#install(FILES "V8-6-0-Energy+.idd" DESTINATION "PreProcess/IDFVersionUpdater")
+#install(FILES "V8-7-0-Energy+.idd" DESTINATION "PreProcess/IDFVersionUpdater")
+#install(FILES "V8-8-0-Energy+.idd" DESTINATION "PreProcess/IDFVersionUpdater")
+#install(FILES "V8-9-0-Energy+.idd" DESTINATION "PreProcess/IDFVersionUpdater")
```
Comment (Myoldmopar) [inline idd/CMakeLists.txt:11]: Not installing the super old IDDs anymore.  We can make sure these are available for the rare user who needs them, but cleaning up the old versions in the installer should save download time and disk space for like 99% of the users.

File: release/Deprecation.in.html [hunk 4]
```diff
@@ -142,18 +135,22 @@ <h3>Geometry Transformation</h3>
     </p>
 
     <h2>Level 2 Deprecation Status</h2>
+
```
Comment (Myoldmopar) [inline release/Deprecation.in.html:138]: Typical readme/documentation updates.

File: src/EnergyPlus/PlantLoopHeatPumpEIR.cc [hunk 5]
```diff
@@ -2365,14 +2365,14 @@ void EIRFuelFiredHeatPump::oneTimeInit(EnergyPlusData &state)
...
-        // SetupOutputVariable(state,
+        // Setup Output Variable(state,
         //                    "Fuel-fired Absorption Heat Pump Source Side Inlet Temperature",
         //                    OutputProcessor::Unit::C,
         //                    this->sourceSideInletTemp,
         //                    OutputProcessor::SOVTimeStepType::System,
         //                    OutputProcessor::SOVStoreType::Average,
         //                    this->name);
-        // SetupOutputVariable(state,
+        // Setup Output Variable(state,
```
Comment (Myoldmopar) [inline src/EnergyPlus/PlantLoopHeatPumpEIR.cc:2375]: The script that scrapes the code building out the output variable file was complaining about the commented SetupOutputVariable calls.  It potentially botches the rudimentary parser.  So just adding spaces so these aren't detected, but they can be quickly fixed back up later if these are enabled.

File: src/Transition/CMakeLists.txt [hunk 6]
```diff
@@ -68,7 +68,7 @@ foreach(i RANGE 1 ${end})
   set_target_properties("${LAST_NAME}" PROPERTIES FOLDER Auxiliary)
   target_link_libraries("${LAST_NAME}" TransitionLib)
 
-  if("${i}" GREATER "22")
+  if("${i}" GREATER "33")
```
Comment (Myoldmopar) [inline src/Transition/CMakeLists.txt:71]: This pushes the install to only include "V9-0 to V9-1" and above.  That change alone could make a notable impact in installer size.

File: workflows/energyplus.py [hunk 7]
```diff
@@ -154,7 +233,8 @@ def eplus_suffixes():
             "Map.txt",
         ]
 
-    def run_energyplus(self, isIP, run_directory, file_name, args, by_api):
+    @staticmethod
```
Comment (Myoldmopar) [inline workflows/energyplus.py:236]: An overall cleanup of this workflow file, not much functional change, mostly refactoring.

File: workflows/transition.py [hunk 8]
```diff
@@ -82,53 +138,43 @@ def get_extra_data(self):
...
-                        message=returned_message,
-                        column_data=[]
-                    )
-                else:
-                    return EPLaunchWorkflowResponse1(
-                        success=False,
-                        message="Transition file not found: {}!".format(''),
-                        column_data=[]
-                    )
+        transition_exes = TransitionWorkflow.find_transition_executable_files(args['workflow location'])
```
Comment (Myoldmopar) [inline workflows/transition.py:141]: Minor refactors in the transition workflow as well.

**Review by Myoldmopar — COMMENTED** [review]
Walkthrough completed, mostly the expected changes with a few cleanups thrown in prior to release.

**Comment by Myoldmopar** [comment]
Assuming CI is happy later today I'll merge this in and tag RC2.  If testing reveals everything is happy we should be releasing on Monday or Tuesday.

**Comment by mjwitte** [comment]
Do you want to address #9874 here?
Update workflows to skip ReadVarsESO if native CSV output is present

File: workflows/energyplus.py [hunk 9]
```diff
@@ -275,13 +337,17 @@ def delete_if_exists(file_path):
         # and at the very end, add the file to run
         command_line_args += [full_file_path]
 
+        # make sure we clean up any existing csv file to detect whether E+ created it natively
```
Comment (Myoldmopar) [inline workflows/energyplus.py:340]: @mjwitte this line deletes the csv if it exists already, then below .....

File: workflows/energyplus.py [hunk 10]
```diff
@@ -301,142 +367,146 @@ def delete_if_exists(file_path):
...
-        mtr_path = os.path.join(output_directory, file_name_no_ext + '.mtr')
-        eplusoutmtr_path = os.path.join(output_directory, 'eplusout.mtr')
-        if os.path.exists(mtr_path):
-            shutil.copy(mtr_path, eplusoutmtr_path)
-
-        if isIP:
-            # run the ConvertESOMTR program to create IP versions of the timestep based output files
+        if os.path.exists(vari_csv_path):
+            # need to skip the read vars stuff, E+ must have created this automatically.
+            pass
```
Comment (Myoldmopar) [inline workflows/energyplus.py:382]: ... this will skip the read vars operations if it does exist.

**Comment by Myoldmopar** [comment]
@mjwitte thanks!  I think #9874 is already fixed here, I had that in my notes when I was mucking around in this stuff, so I just went ahead and took care of it.  I tested it out at least twice, and it seemed to leave the native CSV in place.

**Pull request merged**